File: Cs/Solution/FindIntersectionValues.cs
using System.Collections.Generic;

namespace Solution
{
	public partial class Solution
	{
		public int[] FindIntersectionValues(int[] nums1, int[] nums2)
		{
			HashSet<int> set1 = new();
			HashSet<int> set2 = new();
			int cnt1 = 0, cnt2 = 0;
			foreach (int v in nums1)
			{
				set1.Add(v);
			}
			foreach (int v1 in nums2)
			{
				set2.Add(v1);
			}
			foreach (int v2 in nums1)
			{
				cnt1 += set2.Contains(v2) ? 1 : 0;
			}
			foreach (int v3 in nums2)
			{
				cnt2 += set1.Contains(v3) ? 1 : 0;
			}
			return new int[] { cnt1, cnt2 };
		}
	}
}

File: Ruby/solution/find_intersection_values.rb
# @param {Integer[]} nums1
# @param {Integer[]} nums2
# @return {Integer[]}
def find_intersection_values(nums1, nums2)
  cnt1 = nums1.count { |n| nums2.include?(n) }
  cnt2 = nums2.count { |n| nums1.include?(n) }
  [cnt1, cnt2]
end
